packages/verified-fetch/src/utils/walk-path.ts

```diff
@@ -37,3 +40,23 @@ export async function walkPath (blockstore: ReadableStorage, path: string, optio
 export function isObjectNode (node: UnixFSEntry): node is ObjectNode {
   return node.type === 'object'
 }
+
+/**
+ * Attempts to walk the path in the blockstore, returning ipfsRoots needed to resolve the path, and the terminal element.
+ * If the signal is aborted, the function will throw an AbortError
+ * If a terminal element is not found, a notFoundResponse is returned
+ * If another unknown error occurs, a badGatewayResponse is returned
+ */
+export async function handlePathWalking ({ cid, path, resource, options, blockstore, log }: Omit<FetchHandlerFunctionArg, 'session'> & { blockstore: Blockstore, log: Logger }): Promise<PathWalkerResponse | Response> {
```

Comment:
this was moved to `walk-path.ts` from `verified-fetch.ts` because that file is getting much too large